From 77f9c5edb48b5a68e4940677eb59d35fcf46e04e Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <ra5478@freescale.com>
Date: Thu, 9 Dec 2010 15:34:38 -0600
Subject: [PATCH 546/670] ENGR00136190: MX50 RD1: Incorrect IOMUX settings
 break WVGA display

Setting DSE to high and ODE bits in I2C3_SDA IOMUX pad causes WVGA to fail.
Fix is to use the default values for pad control.

Signed-off-by: Ranjani Vaidyanathan <ra5478@freescale.com>
---
 arch/arm/mach-mx5/mx50_rdp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx5/mx50_rdp.c b/arch/arm/mach-mx5/mx50_rdp.c
index c934483..13fcc9a 100644
--- a/arch/arm/mach-mx5/mx50_rdp.c
+++ b/arch/arm/mach-mx5/mx50_rdp.c
@@ -1239,7 +1239,7 @@ static void fec_gpio_iomux_init()
 			MX50_PAD_I2C3_SDA__GPIO_6_23;
 
 	/* Enable the Pull/keeper */
-	iomux_setting.pad_ctrl = 0x8e;
+	iomux_setting.pad_ctrl = 0x84;
 	mxc_iomux_v3_setup_pad(&iomux_setting);
 	gpio_request(FEC_EN, "fec-en");
 	gpio_direction_output(FEC_EN, 0);
@@ -1255,7 +1255,7 @@ static void fec_gpio_iomux_deinit()
 			MX50_PAD_I2C3_SDA__GPIO_6_23;
 
 	/* Disable the Pull/keeper */
-	iomux_setting.pad_ctrl = 0xE;
+	iomux_setting.pad_ctrl = 0x4;
 	mxc_iomux_v3_setup_pad(&iomux_setting);
 	gpio_request(FEC_EN, "fec-en");
 	gpio_direction_input(FEC_EN);
-- 
1.8.0

